### PR TITLE
feat: render prompts from a template data file

### PR DESCRIPTION
**Pull request by JanDeDobbeleer**
### Prerequisites

- [x] I have read and understood the [contributing guide](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md).
- [x] The commit message follows the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

### Description

Adds a `--data` flag to `print` and `config export image` that renders prompts from a template data file (JSON/YAML/TOML) instead of the live environment, replacing #7644 with a general mechanism.

**The feature**
- `env` section overrides global template properties; `PWD`, `Code`, `ExecutionTime` and `PipeStatus` route into the runtime so live segments see them too. Precedence: CLI flags > data file > live environment.
- `segments` section replays segment writers (keyed by alias, fallback type) exactly like the segment cache does, skipping runtime detection. Any new template property works automatically since both sides share the same reflection round-trip.
- `config export data` records the live context and every enabled segment into a replayable file: record once, replay forever.
- New docs page: `configuration/data`.

**Website pipeline**
- Theme previews render from `website/segment_data.json` — the Kind cluster, `azure/login` and the `id-token` permission are gone from CI.
- The data file shows every configured segment using the fictional names vendors publish for example use (Contoso, Cymbal, AnyCompany, example_corp), with `alice@contoso-devbox` in `~/dev/oh-my-posh`.
- At build time the pipeline injects the current most-played track (Deezer, Apple RSS fallback, explicit-content filtered) into the music segments; any failure falls back to the committed file.

**Image export**
- The canvas is now always exactly `--terminal-width` columns wide (default 120): alignment padding collapses to keep rprompts flush, genuinely long lines wrap like a real terminal, and powerline separators are single width. All 124 theme previews render at an identical 3648px width, so the website finally displays every theme at the same text size.

**Theme fixes**
- chips: removed a redundant `.Env.WAKATIME_API_KEY` template guard.
- tokyo, devious-diamonds: removed references to non-existent properties that errored when the segments render.
- pixelrobots, cloud-context: aliased duplicate `az` segments (cache/toggle key collisions).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/cli/image/image_test.go [hunk 1]
```diff
@@ -1,11 +1,195 @@
...
 import (
+	"os"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+
+	"golang.org/x/image/font"
+	"golang.org/x/image/font/gofont/goregular"
+	"golang.org/x/image/font/opentype"
```
[the remark this answers is not in the document]
Comment (JanDeDobbeleer) [reply src/cli/image/image_test.go:12]: Fixed: the test face now uses gofont/gomono, so a column maps to one consistent advance, matching the monospaced Hack Nerd Font the renderer ships with.

File: src/cli/image/image_test.go [hunk 2]
```diff
@@ -1,11 +1,195 @@
...
 )
 
+// loadTestFont returns a font face backed by the Go Regular TTF bundled with
+// golang.org/x/image, so layout tests don't need network access or the real
+// Hack Nerd Font to measure text.
+func loadTestFont(t *testing.T) font.Face {
+	t.Helper()
+
+	fnt, err := opentype.Parse(goregular.TTF)
```
[the remark this answers is not in the document]
Comment (JanDeDobbeleer) [reply src/cli/image/image_test.go]: Fixed together with the font switch: opentype.Parse now consumes gomono.TTF.

File: src/cli/config_export_image.go [hunk 3]
```diff
@@ -108,6 +120,11 @@ Exports the config to an image file using customized output settings.`,
 			settings.Cursor = "_"
 		}
 
+		// The engine pads right-aligned blocks/rprompts to TerminalWidth, so the
+		// image must clamp to the same column count or the two disagree on
+		// where content should land.
+		settings.Columns = imageTerminalWidth
```
[the remark this answers is not in the document]
Comment (JanDeDobbeleer) [reply src/cli/config_export_image.go:134]: Fixed: the effective width now resolves as --terminal-width > settings file columns > default 120, and is written back to both flags.TerminalWidth and settings.Columns before rendering. TerminalWidth is read lazily at render time, so the engine and canvas stay in sync. Verified: default renders 3648px, a settings file with columns=100 renders 3088px, and --terminal-width=140 on top of that settings file renders 4208px.

File: src/config/data.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	ext := strings.TrimPrefix(strings.ToLower(filepath.Ext(path)), ".")
+
+	var root map[string]json.RawMessage
+
+	switch ext {
+	case JSON, JSONC:
+		if err := json.Unmarshal(raw, &root); err != nil {
+			return nil, fmt.Errorf("failed to parse data file: %w", err)
+		}
+	case YAML, YML:
```
[the remark this answers is not in the document]
Comment (JanDeDobbeleer) [reply src/config/data.go:51]: Fixed: LoadData now strips comments for .json/.jsonc through the same jsonutil.StripComments call the config loader uses, with a commented-JSONC case added to TestLoadDataFormats.

File: website/docs/configuration/data.mdx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      },
+    },
+  }}
+/>
+
+## The `env` section
+
+Keys are the [global template properties][templates] you already use in templates (`PWD`, `AbsolutePWD`,
+`PSWD`, `Folder`, `UserName`, `HostName`, `Shell`, `ShellVersion`, `Root`, `OS`, `WSL`, `SHLVL`, `Jobs`,
+`Code`, `PromptCount`, `Version`, and `Var`), plus `ExecutionTime` and `PipeStatus`.
```
[the remark this answers is not in the document]
Comment (JanDeDobbeleer) [reply website/docs/configuration/data.mdx:52]: Fixed in the docs: the Recording data section now states that the recorder omits Var because the config's own var section already defines those values, and that setting Var in the file by hand still works as an override. The env-key list stays intact since Var is a valid replay key.

File: src/config/segment.go [hunk 6]
```diff
@@ -372,6 +386,31 @@ func (segment *Segment) restoreCache() bool {
...
+	data, OK := segment.env.Flags().SegmentData[segment.DataKey()]
+	if !OK {
+		return false
+	}
+
+	err := json.Unmarshal(data, &segment.writer)
+	if err != nil {
+		log.Error(err)
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (JanDeDobbeleer) [reply src/config/segment.go:402]: This is not what encoding/json does here. Unmarshal into a pointer-to-interface only falls back to map[string]any when the interface is nil or holds a non-pointer; when it holds a non-nil pointer, the decoder follows it and populates the concrete value (see indirect() in encoding/json/decode.go). MapSegmentWithWriter always runs before restoreData, so segment.writer holds the concrete *segments.X at this point. This is also the exact pattern restoreCache has shipped with for years (same file, same &segment.writer target), and TestSegment_RestoreDataPopulatesWriterAndForcesEnabled asserts the concrete-type assertion and the rendered property after replay. Resolving.

File: src/cli/image/image_test.go [hunk 7]
```diff
@@ -1,11 +1,195 @@
...
+	assert.LessOrEqual(t, row.width(), maxWidth, "the collapsed row must fit within the fixed canvas width")
+	assert.Greater(t, row.width(), maxWidth-plan.spaceAdvance, "the collapsed row should end close to the right edge, not far short of it")
+
+	result := glyphString(row.glyphs)
+	assert.True(t, strings.HasPrefix(result, left), "left-aligned content must be preserved")
+	assert.True(t, strings.HasSuffix(result, right), "the right-aligned segment must stay flush against the right edge")
+}
+
+func TestDoubleWidthGlyphNeverSplitsAcrossWrapBoundary(t *testing.T) {
+	icon := '' // Powerline extra symbol range, always double width (see doubleWidthRunes)
```
[the remark this answers is not in the document]
Comment (JanDeDobbeleer) [reply src/cli/image/image_test.go]: Fixed: the test now uses U+F121 from the Font Awesome range, which is still in doubleWidthRunes, so the wrap-boundary edge case is exercised again. Good catch after the Powerline range moved to single width.

File: website/export_themes.mjs [hunk 8]
```diff
@@ -70,6 +85,137 @@ function getThemeNameFromFile(fileName) {
...
+ */
+async function trendingFromAppleRSS() {
+  const body = await fetchJsonWithTimeout('https://rss.marketingtools.apple.com/api/v2/us/music/most-played/25/songs.json');
+  const tracks = body?.feed?.results;
+
+  if (!Array.isArray(tracks) || tracks.length === 0) {
+    throw new Error('no tracks returned');
+  }
+
+  // Apple only includes contentAdvisoryRating (literal value "Explict") when a track is explicit,
```
[the remark this answers is not in the document]
Comment (JanDeDobbeleer) [reply website/export_themes.mjs]: That misspelling is Apple's, not ours: the live feed returns the literal string "Explict" (verified today, 13 flagged tracks, all with that exact value). The code filters on key presence so the value never matters, but the comment now spells this out with (sic) so it doesn't get corrected into being wrong.
